src/inmanta/data.py

```diff
@@ -1130,34 +1130,62 @@ class Report(BaseDocument):
         :param outstream what was reported on system out
     """
 
-    id = Field(field_type=uuid.UUID, required=True, part_of_primary_key=True)
-    started = Field(field_type=datetime.datetime, required=True)
-    completed = Field(field_type=datetime.datetime, required=True)
-    command = Field(field_type=str, required=True)
-    name = Field(field_type=str, required=True)
-    errstream = Field(field_type=str, default="")
-    outstream = Field(field_type=str, default="")
-    returncode = Field(field_type=int)
-    compile = Field(field_type=uuid.UUID)
+    id: uuid.UUID = Field(field_type=uuid.UUID, required=True, part_of_primary_key=True)
+    started: datetime.datetime = Field(field_type=datetime.datetime, required=True)
+    completed: datetime.datetime = Field(field_type=datetime.datetime)
+    command: str = Field(field_type=str, required=True)
+    name: str = Field(field_type=str, required=True)
+    errstream: str = Field(field_type=str, default="")
+    outstream: str = Field(field_type=str, default="")
+    returncode: int = Field(field_type=int)
+    compile: uuid.UUID = Field(field_type=uuid.UUID, required=True)
+
+    async def update_streams(self, out="", err="") -> None:
+        if not out and not err:
+            return
+        await self._execute_query(
+            f"UPDATE {self.table_name()} SET outstream = outstream || $1, errstream = errstream || $2 WHERE id = $3",
```

Comment:
Should we check that out and err end with a new line? and otherwise add it?

Reply:
there is guarantee that chunks read from the stream end on a newline.
This way, we faithfully store the data as it is reported, without adding (or removing) newlines.